src/Gui/Utils/KeyMappingManager.hpp

```diff
@@ -58,23 +58,55 @@ class RA_GUI_API KeyMappingManager : public Ra::Core::Utils::ObservableVoid
-    /// @param context the context of the action
-    /// @param keyString  represents the key that needs to be pressed to trigger the event
+    /// \param context the context of the action
+    /// \param keyString  represents the key that needs to be pressed to trigger the event
     /// (ie Key_Z, for example), "" or "-1" corresponds to no key needed.
-    /// @param modifiersString represents the modifier used along with key or mouse button `
+    /// \param modifiersString represents the modifier used along with key or mouse button `
     /// (needs to be a Qt::Modifier enum value) to trigger the action. Multiples modifiers can be
     /// specified, separated by commas as in "ControlModifier,ShiftModifier".
-    /// @param buttonsString represents the button to trigger the event (e.g. LeftButton).
-    /// @param wheelString if true, it's a wheel event !
-    /// @param actionString represents the KeyMappingAction enum's value you want to
+    /// \param buttonsString represents the button to trigger the event (e.g. LeftButton).
+    /// \param wheelString if true, it's a wheel event !
+    /// \param actionString represents the KeyMappingAction enum's value you want to
     /// trigger.
+    /// \param saveToConfigFile set it to true (the default) to save the action on the config file.
+
     void addAction( const std::string& context,
                     const std::string& keyString,
                     const std::string& modifiersString,
                     const std::string& buttonsString,
                     const std::string& wheelString,
-                    const std::string& actionString );
+                    const std::string& actionString,
+                    bool saveToConfigFile = true );
+
+    /// Add a given action to the mapping system.
+    /// This allow to define custom behavior when some KeyMappingManageable object is not
+    /// parameterized in the application config file.
+    /// This method do not add the action in the config file
+    /// @param context the context of the action.
+    /// @param actionName represents the KeyMappingAction enum's value you want to
+    /// trigger.
+    /// @param keyString  represents the key that needs to be pressed to trigger the event
+    /// (ie Key_Z, for example), "" or "-1" corresponds to no key needed.
+    /// @param modifiersString represents the modifier used along with key or mouse button `
+    /// (needs to be a Qt::Modifier enum value) to trigger the action. Multiples modifiers can be
+    /// specified, separated by commas as in "ControlModifier,ShiftModifier".
+    /// @param buttonsString represents the button to trigger the event (e.g. LeftButton).
+    /// @param wheelString if true, it's a wheel event !
+    /// @return an invalid action if context is not valid, or if actionName has  not been created.
+    // (i,e action.isInvalid())
+    KeyMappingAction addAction( const Context& context,
```

Comment:
The new functions allow to be more concise when creating a new (non existing) custom context with some actions and follow the principle : 
 1. Create the context and check for validity (valid if no context with the same name exists)
 2. if the context is valid, Add actions to the existing context given its id

The existing ones need to be more verbose as they create both the context and the action in the same time without returning their id and requires to do the following : 
 1. Check for existance of the context name
 2. If the context do not exist, create actions given the context name and action name (this will create the context while creating the first action)
 3. Get the context id
 4. For each action, get the action id.

In the demo, creating a context with 2 actions using the added API takes 4 lines while it takes 8 with the existing API. 
In this PR, I also add the addContext method which allows to do the concise version above. I keep the old version to be backward compatible with existing code. 
The verbose version, however, is relevant when reading configs from a xml file while the concise one are relevant when adding custom context and actions and even allows to do that dynamically.

I agree that the doc should be updated but we can also discuss on refactoring the KeyMappingManager API to allow more user friendly usage of the functionalities.